plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1930,6 +1955,19 @@ public void setTableProperties(ConnectorSession session, ConnectorTableHandle ta
         }
     }
 
+    private MetadataEntry buildMetadataEntry(MetadataEntry metadataEntry, Map<String, String> configuration, long createdTime)
+    {
+        return new MetadataEntry(
+                metadataEntry.getId(),
+                metadataEntry.getName(),
+                metadataEntry.getDescription(),
+                metadataEntry.getFormat(),
+                metadataEntry.getSchemaString(),
+                metadataEntry.getOriginalPartitionColumns(),
+                ImmutableMap.copyOf(configuration),
```

Comment:
no need to copyOf here. it's MetadataEntry constructor's responsibility to do any defensive copies it needs